cli/toolchain_cmd.go

```diff
@@ -740,3 +738,45 @@ func toolchainFQN(toolchainName string) string {
 func toolchainCloneURI(toolchainName string) string {
 	return "https://github.com/sourcegraph/" + toolchainName
 }
+
+// checkPythonVersion ensures we have Python 3.5 installed.
+// On Windows, Python 3.5 does not have python3.5.exe,
+// so will run `python --version` to check it
+func checkPythonVersion() error {
+
+	if runtime.GOOS != "windows" {
+		return nil
+	}
+
+	cmd := exec.Command("python", "--version")
+	out, err := cmd.Output()
+
+	if err != nil {
+		return fmt.Errorf("unable to detect Python version")
+	}
+
+	version := regexp.MustCompile("^Python (\\d+)\\.(\\d+)").FindStringSubmatch(string(out))
+	if version == nil {
+		return fmt.Errorf("unable to detect Python version from version string '%s'", string(out))
+	}
+
+	major, err := strconv.Atoi(version[1])
```

Comment:
It shouldn't. `version` keeps matching string and all the sub-expressions (re above defines 2 of them). If there is no match, `version` will be `nil`